src/panels/alarm-configuration/alarmParser.tsx

```diff
@@ -0,0 +1,55 @@
+import { DataFrame, FieldCache, DataQueryRequest } from '@grafana/data';
+import { AlarmInfo, AlarmResultFields, AlarmQueryInfo } from './types';
+import { TwinMakerQuery } from 'common/manager';
+
+export function processAlarmQueryInput(query: DataQueryRequest<TwinMakerQuery> | undefined): AlarmQueryInfo {
+  const info: AlarmQueryInfo = {};
+
+  if (!query) {
+    info.invalidFormat = true;
+    info.warning = 'No query to parse';
+  }
+
+  const target = query?.targets[0];
+  info.alarmName = target?.componentName;
+  info.entityId = target?.entityId;
+
+  return info;
+}
+
+export function processAlarmResult(data: DataFrame[]): AlarmInfo {
+  const info: AlarmInfo = {};
+
+  if (!data?.length) {
+    info.invalidFormat = true;
+    info.warning = 'missing data';
+    return info;
+  }
+
+  // latest only verison
+  for (const frame of data) {
+    const cache = new FieldCache(frame);
```

Comment:
I'm not sure what benefit this has, in component form the state manager can handle this and limit the updates, effectively not needing any caching stuff to have the same behavior.

Reply:
Not doing it for the caching but for the accessor functions that Grafana adds to the FieldCache class over their raw frame class.